packages/runtime/src/controls/control.ts

```diff
@@ -195,3 +214,58 @@ export function getTranslatableData(definition: Descriptor | ControlDefinition,
       return null
   }
 }
+
+export function mergeTranslatedData(
+  definition: PropControllerDescriptor,
+  data: Data,
+  translatedData: Data,
+  context: MergeTranslatableDataContext,
+): Data {
+  switch (definition.type) {
+    case Types.TextInput:
+    case Types.TextArea:
+    case TextInputControlType:
+    case TextAreaControlType:
+      if (translatedData == null) return data
+
+      return translatedData
```

Comment:
For multi-line switch blocks, I recommend using a block (`{}`) to make it easer to parse visually